### PR TITLE
add script to plot sensitivity

### PR DESCRIPTION
**Pull request by vuillaut**
To add in workflow:
- produce the files as f'{prod_id}_sensitivity.fits.gz'
- produce plots 

in IRF dir

### CODE REVIEW

File: lstmcpipe/plot_irfs.py [hunk 1]
```diff
@@ -0,0 +1,408 @@
+from astropy.table import QTable
+import matplotlib.pyplot as plt
+import numpy as np
+import astropy.units as u
+from pyirf.utils import cone_solid_angle
```
Comment (garciagenrique) [inline lstmcpipe/plot_irfs.py:5]: Haven't work with pyirf a lot. Any incompatibilities between lstchain and pyirf ? should we specify the version for the installing

Comment (garciagenrique) [reply lstmcpipe/plot_irfs.py:5]: Also @vuillaut, should we move the script to the `plot` package of the module ?

Comment (maxnoe) [reply lstmcpipe/plot_irfs.py:5]: Lstchain also uses pyirf and there shouldn't be a proble

**Comment by garciagenrique** [comment]
Hi @vuillaut,
The PR is ready. However when running the `dl2_to_sensitivity` for `gamma-diffuse` particles I get the following error. Any idea ? 

```
(lst-dev) enrique.garcia@cp01 lstmcpipe (sensitivity) $ lstmcpipe_dl2_to_sensitivity -g /fefs/aswg/workspace/enrique.garcia/workflow_r0_dl2_lst/DL2/20200629_prod5_trans_80/gamma-diffuse/zenith_20deg/south_pointing/20210716_v0.7.3_prod5_trans_80_v00/dl2_gamma-diffuse_20deg_180deg_20210716_v0.7.3_prod5_trans_80_v00_testing.h5 -p /fefs/aswg/workspace/enrique.garcia/workflow_r0_dl2_lst/DL2/20200629_prod5_trans_80/proton/zenith_20deg/south_pointing/20210716_v0.7.3_prod5_trans_80_v00/dl2_proton_20deg_180deg_20210716_v0.7.3_prod5_trans_80_v00_testing.h5 -e /fefs/aswg/workspace/enrique.garcia/workflow_r0_dl2_lst/DL2/20200629_prod5_trans_80/electron/zenith_20deg/south_pointing/20210716_v0.7.3_prod5_trans_80_v00/dl2_electron_20deg_180deg_20210716_v0.7.3_prod5_trans_80_v00_testing.h5  -o /fefs/aswg/workspace/enrique.garcia/workflow_r0_dl2_lst/IRF/20200629_prod5_trans_80/zenith_20deg/south_pointing/20210716_v0.7.3_prod5_trans_80_v00/diffuse/20210716_v073_prod5_trans_80_v00_gamma_diffuse_sensitivity.fits.gz
INFO:lstchain MC DL2 to IRF - sensitivity curves:Simulated Gamma Events:
===== gamma =====
Traceback (most recent call last):
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/quantity.py", line 738, in to_value
    scale = self.unit._to(unit)
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/core.py", line 951, in _to
    f"'{self!r}' is not a scaled version of '{other!r}'")
astropy.units.core.UnitConversionError: 'Unit("m2 sr / cm2")' is not a scaled version of 'Unit(dimensionless)'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/bin/lstmcpipe_dl2_to_sensitivity", line 8, in <module>
    sys.exit(main())
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/lstmcpipe/stages/script_dl2_to_sensitivity.py", line 159, in main
    p["events"]["true_energy"], p["target_spectrum"], p["simulated_spectrum"]
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/decorators.py", line 251, in wrapper
    return_ = wrapped_function(*func_args, **func_kwargs)
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/pyirf/spectral.py", line 59, in calculate_event_weights
    u.one
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/quantity.py", line 741, in to_value
    value = self._to_value(unit, equivalencies)
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/quantity.py", line 661, in _to_value
    equivalencies=equivalencies)
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/core.py", line 987, in to
    return self._get_converter(other, equivalencies=equivalencies)(value)
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/core.py", line 918, in _get_converter
    raise exc
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/core.py", line 904, in _get_converter
    self, other, self._normalize_equivalencies(equivalencies))
  File "/fefs/aswg/workspace/enrique.garcia/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/core.py", line 888, in _apply_equivalencies
    unit_str, other_str))
astropy.units.core.UnitConversionError: 'm2 sr / cm2' (solid angle) and '' (dimensionless) are not convertible
```

**Comment by LukasNickel** [comment]
I believe the script only works for pointlike gammas.
It tries to calculate the weights for a pointlike Crab spectrum, which makes no sense for diffuse gammas.
Also, sensitivites are "always" for a pointlike observation if I am not mistaken. Are you sure that you want to calculate a sensitivity with diffuse gammas as source?

**Comment by maxnoe** [comment]
> Also, sensitivites are "always" for a pointlike observation if I am not mistaken

Yes, this is always (without quotes) point-like sensitivity

**Comment by garciagenrique** [comment]
Perfect, thanks for the clarification guys !

**Comment by garciagenrique** [comment]
Also, while trying to batch the plotting part (`plots.plot_irfs`) I got the following qt error (running interactively works, though)

```
qt.qpa.xcb: could not connect to display localhost:16.0
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, minimal, minimalegl, offscreen, vnc, webgl, xcb.
```

I have found [this solution](https://stackoverflow.com/questions/60042568/this-application-failed-to-start-because-no-qt-platform-plugin-could-be-initiali), but not couldn't test it yet.

**Comment by maxnoe** [comment]
Don't you run it with `MPLBACKEND=Agg`?

**Comment by garciagenrique** [comment]
> Don't you run it with `MPLBACKEND=Agg`?

Ah... haven't tried indeed. I thought the declaration of the backend was only matplotlib-related. Thanks again for the tip !

**Comment by maxnoe** [comment]
This is a matplotlib error as far as I can see (matplotlib trying to use the QT backend for interactive plots)

**Comment by garciagenrique** [comment]
Thanks Max, indeed the problem was not defining the backend.

The PR is ready. It produces the following output in the `/IRF` directory (taking out `*.e` `*.o`  and copied config files):

```
.
|____off0.0deg
| |____onsite_mc_dl2_to_irfs.py
| |____IRFFITSWriter.provenance.log
| |____irf_20210718_v073_prod5_trans_80_v00_gamma_point-like_off00deg.fits.gz
| |____20210718_v073_prod5_trans_80_v00_gamma_off00deg_sensitivity.fits.gz
| |____20210718_v073_prod5_trans_80_v00_gamma_off00deg_sensitivity.png
|____off0.4deg
| |____onsite_mc_dl2_to_irfs.py
| |____IRFFITSWriter.provenance.log
| |____irf_20210718_v073_prod5_trans_80_v00_gamma_point-like_off04deg.fits.gz
| |____20210718_v073_prod5_trans_80_v00_gamma_off04deg_sensitivity.fits.gz
| |____20210718_v073_prod5_trans_80_v00_gamma_off04deg_sensitivity.png
|____diffuse
| |____onsite_mc_dl2_to_irfs.py
| |____IRFFITSWriter.provenance.log
| |____irf_20210718_v073_prod5_trans_80_v00_gamma_diffuse.fits.gz
```

**Pull request merged**